File: backend/src/main/java/com/enterprise/meeting/repository/MemoRepository.java
package com.enterprise.meeting.repository;

import com.enterprise.meeting.entity.Memo;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;
import java.util.List;

public interface MemoRepository extends JpaRepository<Memo, Long> {
    List<Memo> findByUserIdOrderByCreatedAtDesc(Long userId);

    Page<Memo> findByUserIdAndDeletedAtIsNullOrderByCreatedAtDesc(Long userId, Pageable pageable);

    @Query("SELECT m FROM Memo m WHERE m.user.id = :userId AND m.isCompleted = false AND m.deletedAt IS NULL ORDER BY m.createdAt DESC")
    List<Memo> findIncompleteByUser(@Param("userId") Long userId);
}

File: android/app/src/main/java/com/enterprise/meeting/domain/repository/MemoRepository.kt
package com.enterprise.meeting.domain.repository

import com.enterprise.meeting.domain.model.Memo

interface MemoRepository {
    suspend fun getMemos(userId: Long): Result<List<Memo>>
    suspend fun createMemo(
        userId: Long,
        title: String,
        content: String?,
        reminderAt: Long?,
        color: String?
    ): Result<Memo>
    suspend fun updateMemo(
        id: Long,
        title: String?,
        content: String?,
        isCompleted: Boolean?
    ): Result<Memo>
    suspend fun deleteMemo(id: Long): Result<Unit>
}
